### ISSUE TITLE
Cop idea: assign variable directly to result of if/unless block

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#12968]
I often encounter a pattern where developers set some variable to `nil`, and then set it again inside an `if/unless` block. The same can be achieved by simply assigning the result of the `if` block directly to the variable.

```ruby
# bad
result = nil
if cond?
  result = foo(...)
end

# good
result = if cond?
  foo(...)
end
```

**Comment by Earlopain** [comment]
You don't even need to have the variable outside, you can also do this:

```rb
if cond?
  result = foo(...)
end
```

Regardless of if the branch is entered, the `result` variable will be defined after. Though I agree that the second one looks cleaner

**Comment by koic** [comment]
Can you first open a proposal in the style guide to discuss the merits and demerits of this style?
https://github.com/rubocop/ruby-style-guide
